docs/README.md

```diff
@@ -74,14 +74,25 @@ The following would only render (and remove) the error messages:
 <?= $this->Flash->render('flash', ['types' => ['error']]) ?>
 ```
 
+### Use in AJAX requests
+
+Set the request header `X-Get-Flash: yes` in your AJAX calls to receive a json encoded response in the `X-Flash` with an array of keys containing an array of messages (each flash message contains a message key, a type key and params key):
```

Comment:
instead of yes: Should we use
- `all` (both session and transient), we could also keek `yes` or `on`
- `transient`
- `session`

as options?
Usually only "transient" make really sense, as those are generated by guarantee only in this request (no side effects).

Reply:
Also, if transient is the default, then you dont need the header, that one you only need if you want to modify to include or use session. :) So everyone is happy.

Reply:
will consuming the transient messages in the component take them away from ajax view flash rendering? we wouldn't want the default to stop the messages reaching the view

Reply:
> Usually only "transient" make really sense for AJAX
Sometimes you use non-transient messages within controller methods that can be called ajax and normal. But this is not a bad options.... we can skip this whole request header in favour of recommending the use of transient messages

Reply:
Either way works for me.